### PR TITLE
V4P3: refactor: Move most Venia components to separate UI library

### LINKED ISSUE #1551: [feature]: Split Venia codebase into top-layer app and component library
The Venia codebase consists of both reusable component code and fully integrate app code. Most implementors will want to use the former as a dependency, while managing their own copy of the latter. The current Venia codebase contains code that would never be imported as a dependency. Creating a new project requires either using fallback configuration or copying the entirety of Venia.

**Describe the solution you'd like**
Separate Venia into an app package, and a UI library package.

**Additional context**
This is part of the requirements surrounding project creation and scaffolding. Relevant tickets:
#946
#947 
#1497 
#1498 
#1499 
#1500 
#1501 

**Please let us know what packages this feature is in regards to:**
- [x] `venia-concept`
- [ ] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`

### PR DESCRIPTION
**Pull request by zetlen**
## Description

Move most Venia UI into a consumable library, `@magento/venia-ui`. Reduce `@magento/venia-concept` footprint down to a "shell" which imports and presents the `App` object from `@magento/venia-ui`.

This decoupling allows for scaffolding (in an upcoming PR) to produce a very small starter project, without copying a huge number of files into the fresh project directory for the developer to maintain.

- Add `venia-ui` workspace.
- Copy most Venia code into `venia-ui` workspace.
- Change the files remaining in `venia-concept` to import their components from `venia-ui`.
- Break `upward.yml` into an un-opinionated piece of boilerplate in `venia-concept`, and all the Venia-specific implementations in `venia-ui/upward.yml`. Use new `UpwardIncludePlugin` to merge UPWARD files and add their dependencies to the build.

## Verification Steps

1. From a freshly cloned repo, run `yarn run stats:venia`.
2. When the tree diagram of dependencies appears, note the amount of code contained in node_modules versus in the local app. It should be much less.

## Related Issues
#946 and #947
Closes #1551.

## Proposed Labels for Change Type/Package

- [x] major (e.g x.0.0 - a breaking change)
- [ ] minor (e.g 0.x.0 - a backwards compatible addition)
- [ ] patch (e.g 0.0.x - a bug fix)

## Checklist:

- [ ] I have updated the documentation accordingly, if necessary.
- [ ] I have added tests to cover my changes, if necessary.

**I'd like to expedite this if possible, because it's a lot of renames and merges will get hairy if we leave it out too long.**

### CODE REVIEW

File: packages/venia-concept/webpack.config.js [hunk 1]
```diff
@@ -22,15 +22,16 @@ module.exports = async env => {
         ],
         special: {
             '@magento/peregrine': {
-                esModules: true
+                esModules: true,
+                cssModules: true
```
Comment (jimbo) [inline packages/venia-concept/webpack.config.js]:
```suggestion
```

File: packages/venia-concept/package.json [hunk 2]
```diff
@@ -87,7 +93,7 @@
     "prettier": "~1.16.4",
     "prop-types": "~15.7.2",
     "react": "~16.8.3",
-    "react-apollo": "~2.5.5",
+    "react-apollo": "2.5.5",
```
Comment (jimbo) [inline packages/venia-concept/package.json]: Good idea. 👍

**Comment by dpatil-magento** [comment]
@zetlen 
1: Getting data fetch error on accessing categories try https://pr-1499.pwa-venia-staging.com/venia-tops.html
2. There is test failure packages/venia-concept/webpack.config.js  `prettier`

**Comment by zetlen** [comment]
@dpatil-magento That was caused by an incompatibility in `react-apollo` module versions. I've resolved it and pushed the change.

**Comment by dpatil-magento** [comment]
@zetlen QA Pass but needs reviewer approval on recent changes.

**Comment by wokakira** [comment]
Why does classify.js exist in /lib/classify.js vs lib/util/classify.js

**Pull request merged**